application/change-organizations.mjs

```diff
@@ -34,18 +35,19 @@ export default async function changeOrganizations (institutions, organizations)
     if (organizations.some(hasName)) {
       const organization = organizations.find(hasName)
       const institutionName = getInstitutionName(institution)
+      const targetInstitutionMetaData = createMetaData(institution)
       const organizationMetaData = getMetadata(organization)
 
       if (
-        institutionName !== getOrganizationDisplayName(organization) ||!compareMetaDataValue(organizationMetaData, 'institutionId', institutionId)) {
+        institutionName !== getOrganizationDisplayName(organization) || hasChangedMetaData(organizationMetaData, targetInstitutionMetaData)) {
         const {
           id,
           ...rest
         } = organization
 
         let status
         try {
-          status = await updateOrganizationById(id, { ...rest, name: institutionId, display_name: institutionName, metadata: { ...organization.metadata, institutionId } })
+          status = await updateOrganizationById(id, { ...rest, name: institutionId, display_name: institutionName, metadata: targetInstitutionMetaData })
```

Comment:
I think we should instead do:

{
... organisationMetadata ?? {},
... targetInstitutionMetadata
} 

not to override anything that is already there but with different metadata keys, perhaps added manually or by another team.

EDIT: (this should perhaps be done in the createMetaData function since the same logic applies in the other file).